site/_posts/2020-07-23-securing-react-native-with-oauth.md

```diff
@@ -0,0 +1,480 @@
+
+Now, we can move on to coding.
+
+## The implementation
+
+As always, if you want to skip ahead, grab the code from the [GitHub repository](https://github.com/fusionauth/fusionauth-example-react-native).
+
+First, we need to modify the `App.js` file. Add necessary imports as shown in the code snippet below:
+
+```javascript
+//...
+import React, { useState, useCallback, useMemo } from 'react';
+import { Alert } from 'react-native';
+import { authorize, refresh, revoke, prefetchConfiguration } from 'react-native-app-auth';
+//...
+```
+
+Next, we need to create a `configs` object as shown in the code snippet below:
+
+```javascript
+//...
+const configs = {
+  fusionauth: {
+    issuer: 'https://ce25267ff5a5.ngrok.io', 
+    clientId: '253eb7aa-687a-4bf3-b12b-26baa40eecbf',
+    redirectUrl: 'fusionauth.demo:/oauthredirect',
+    additionalParameters: {},
+    scopes: ['offline_access'],
+  }
+}
+//...
+```
+
+Here's more information about the properties of the `fusionauth` object in the `configs` object:
+
+* `issuer` is the URL for the FusionAuth server.
+* `clientId` is the ID that we grabbed from the FusionAuth dashboard.
+* `redirectURL` is the URL that we set up before, with a callback path defined by the react native app auth library.
+
+Make sure you update the `issuer` and `clientId` keys in this object with your configuration values.
+
+We also add any additional parameters we need to pass (none, in this case), as well as scopes to request. We're requesting the `offline_access` scope as that will return a `refresh_token`, which can be used to request additional access tokens.
+
+Then, create a default auth state in order to handle the response from the server, as shown in the code snippet below:
+
+```javascript
+//...
+const defaultAuthState = {
+  hasLoggedInOnce: false,
+  provider: '',
+  accessToken: '',
+  accessTokenExpirationDate: '',
+  refreshToken: ''
+};
+
+const [authState, setAuthState] = useState(defaultAuthState);
+//...
+```
+
+Now, we are ready to configure authorization.
+
+## Configuring Authorization
+
+We need to create a function to perform authorization. This will require the `configs` object previously created. The implementation of the function is below:
+
+```javascript
+//...
+const handleAuthorize = useCallback(
+  async provider => {
+    try {
+      const newAuthState = await authorize(configs.fusionauth);
+
+      setAuthState({
+        hasLoggedInOnce: true,
+        ...newAuthState
+      });
+    } catch (error) {
+      Alert.alert('Failed to log in', error.message);
+    }
+  },
+  [authState]
+);
+//...
+```
+
+## Building the user interface
+
+Next, we need to create a user interface (UI). We'll create a simple UI to display the access token and any other server response data. The access token is what FusionAuth provides once a user has authenticated. The code for UI implementation looks like this:
+
+```react
+//...
+return (
```

Comment:
No clue where this code goes either. I think the author is assuming that everyone knows a lot more about React Native or even the specific app in question than a beginner would.